static/js/stream_popover.js

```diff
@@ -327,6 +328,19 @@ exports.register_stream_handlers = function () {
         e.stopPropagation();
     });
 
+    // Toggle displaying starred message count
+    $('body').on('click', '#toggle_display_starred_msg_count', function (e) {
+        exports.hide_starred_messages_popover();
+        e.preventDefault();
+        e.stopPropagation();
+        var starred_message_counts = page_params.starred_message_counts;
+        var data = {};
+        data.starred_message_counts = JSON.stringify(!starred_message_counts);
+
+        // do_settings_change expects one jQuery element to display success status. We passed
+        // '.left-sidebar-modal-holfer' here. We can create a new one, but that makes no sense
+        settings_ui.do_settings_change(channel.patch, '/json/settings/display', data, '.left-sidebar-modal-holder');
```

Comment:
Thanks for working on this :)

> In this case we don't want to display any success message on successful toggling.

If this is the case, then yeah definitely, it doesn't make sense to me as well to use `do_settings_change` function, it should only be used when you expect to show that nice "Saving.."/"Saved" mechanism feedback. You should directly use `channel.patch` (you can grep for that) and just pass `url` and `data`.

Reply:
Thanks! "Changing data on backend will send event to update frontend" was the part I missed here :). Earlier what I was afraid of was, not doing setting change will not update in settings section. Updated corresponding line, and now it just uses `channel.patch`. Thanks again :)